File: leetcode/2- Add Two Numbers/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(A *ListNode, B *ListNode) *ListNode {
	R := &ListNode{}
	L := R
	carry := 0
	for A != nil || B != nil || carry > 0 {
		var a, b int = 0, 0
		if A != nil {
			a = A.Val
		}
		if B != nil {
			b = B.Val
		}

		s := a + b + carry
		carry = s / 10

		L.Next = &ListNode{Val: s % 10, Next: nil}
		L = L.Next

		if A != nil {
			A = A.Next
		}
		if B != nil {
			B = B.Next
		}
	}
	return R.Next
}

File: leetcode/2- Add Two Numbers/solution.py
class Solution:
    def addTwoNumbers(self, A: ListNode, B: ListNode) -> ListNode: 
        carry = 0
        r = ListNode(0)
        L = r
        while A or B or carry:
            a = A.val if A else 0
            b = B.val if B else 0
            carry, s = divmod(a + b + carry, 10)
            
            L.next = ListNode(s)
            L = L.next
            

            A = None if not A else A.next
            B = None if not B else B.next

        return r.next